### PR TITLE
feat(processors.template): Allow `tag` to be a template

### PR DESCRIPTION
**Pull request by srebhan**
- [x] Updated associated README.md.
- [x] Wrote appropriate unit tests.
- [x] Pull request title or commits are in [conventional commit format](https://www.conventionalcommits.org/en/v1.0.0/#summary)

This PR allows the `tag` option to also be a Go template to make it more flexible.

### CODE REVIEW

File: plugins/processors/template/README.md [hunk 1]
```diff
@@ -48,6 +50,19 @@ See the [CONFIGURATION.md][CONFIGURATION.md] for more details.
...
+
+```toml
+[[processors.template]]
+  tag = '{{ .Field "type" }}'
+  template = '{{ .Name }}'
+```
+
+```diff
+- cpu,level=debug,hostname=localhost time_idle=42,type=sensor
++ cpu,level=debug,hostname=localhost,sensor=abc123xyz time_idle=42,type=sensor
```
Comment (Hipska) [inline plugins/processors/template/README.md:63]: If I'm correct, the `{{.Name}}` would result in `cpu` isn't it?

**Review by Hipska — APPROVED** [review]

File: plugins/processors/template/README.md [hunk 2]
```diff
@@ -24,12 +24,14 @@ See the [CONFIGURATION.md][CONFIGURATION.md] for more details.
 ```toml @sample.conf
 # Uses a Go template to create a new tag
 [[processors.template]]
-  ## Tag to set with the output of the template.
+  ## Go template used to create the tag name of the output. In order to
+  ## ease TOML escaping requirements, you should use single quotes around
+  ## the# template string.
```
Comment (powersj) [inline plugins/processors/template/README.md:29]: extra #?

```suggestion
  ## the template string.
```

File: plugins/processors/template/README.md [hunk 3]
```diff
@@ -24,12 +24,14 @@ See the [CONFIGURATION.md][CONFIGURATION.md] for more details.
...
   tag = "topic"
 
-  ## Go template used to create the tag value.  In order to ease TOML
-  ## escaping requirements, you may wish to use single quotes around the
-  ## template string.
+  ## Go template used to create the tag value of the output. In order to
+  ## ease TOML escaping requirements, you should use single quotes around
+  ## the# template string.
```
Comment (powersj) [inline plugins/processors/template/README.md:34]:
```suggestion
  ## the template string.
```

File: plugins/processors/template/sample.conf [hunk 4]
```diff
@@ -1,9 +1,11 @@
 # Uses a Go template to create a new tag
 [[processors.template]]
-  ## Tag to set with the output of the template.
+  ## Go template used to create the tag name of the output. In order to
+  ## ease TOML escaping requirements, you should use single quotes around
+  ## the# template string.
```
Comment (powersj) [inline plugins/processors/template/sample.conf:5]:
```suggestion
  ## the template string.
```

File: plugins/processors/template/sample.conf [hunk 5]
```diff
@@ -1,9 +1,11 @@
...
   tag = "topic"
 
-  ## Go template used to create the tag value.  In order to ease TOML
-  ## escaping requirements, you may wish to use single quotes around the
-  ## template string.
+  ## Go template used to create the tag value of the output. In order to
+  ## ease TOML escaping requirements, you should use single quotes around
+  ## the# template string.
```
Comment (powersj) [inline plugins/processors/template/sample.conf:10]:
```suggestion
  ## the template string.
```

**Review by powersj — APPROVED** [review]

**Review by powersj — APPROVED** [review]

**Pull request merged**